minifi/minifi-nar-bundles/minifi-framework-bundle/minifi-framework/minifi-resources/src/main/resources/conf/logback.xml

```diff
@@ -35,6 +35,8 @@
             <maxFileSize>1MB</maxFileSize>
             <!-- Provide a cap of 10 MB across all archive files -->
             <totalSizeCap>10MB</totalSizeCap>
+            <!-- This is supposed to force log cleanup when the service starts/restarts -->
+            <cleanHistoryOnStart>true</cleanHistoryOnStart>
```

Comment:
what does this mean?  it cleans everything or just to what the limit was?   If i keep 30 files or 30 days does it clean all that or does it get anything older?  Why is that needed if that is what it is does?

Reply:
When I tested it locally, I set `maxHistory` to 1 so I could easily experiment with logback's behavior. I found the cleanup was really inconsistent, but if I set `cleanHistoryOnStart` to true it would at least be (or seem to be) very strict at enforcing cleanup if the node got bounced. The workflow I did was:

1. Set maxHistory to 1 (1 day)
2. Copy the current timestamped file to TS - 1 day
3. Restart node

On restart, logback cleaned up the old files.

Reply:
I tested this briefly, and cleanHistoryOnStart did respect the existing maxHistory when it cleaned up much older files that were outside that window.